python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1510,6 +1506,26 @@ def get_run():
         except DagsterRunAlreadyExists:
             return get_run()
 
+    def handle_started_run(self, run_record: RunRecord, logger: logging.Logger) -> None:
+        max_time_str = run_record.dagster_run.tags.get(MAX_RUNTIME_TAG)
+
+        # If MAX_RUNTIME_TAG has not been set, nothing to do
+        if not max_time_str:
+            return
+        max_time = float(max_time_str)
+        if (
+            run_record.start_time is not None
+            and pendulum.now().timestamp() - run_record.start_time > max_time
```

Comment:
is this just hipster `time.time()`

Reply:
if you use pendulum i believe you want pendulum.now("UTC") (not positive if that matters if you then call .timestamp() on it)

Reply:
pendulum is useful for testing bc you can use `pendulum.test` to freeze `now`